app/http/httpclient.c

```diff
@@ -200,16 +200,29 @@ static void ICACHE_FLASH_ATTR http_connect_callback( void * arg )
 
 	char buf[69 + strlen( req->method ) + strlen( req->path ) + strlen( req->hostname ) +
 		 strlen( req->headers ) + strlen( post_headers )];
-	int len = os_sprintf( buf,
-			      "%s %s HTTP/1.1\r\n"
-			      "Host: %s:%d\r\n"
-			      "Connection: close\r\n"
-			      "User-Agent: ESP8266\r\n"
-			      "%s"
-			      "%s"
-			      "\r\n",
-			      req->method, req->path, req->hostname, req->port, req->headers, post_headers );
 
+	if ((req->port == 80) || (req->port == 443))
```

Comment:
I did miss there was already a req->secure field in the structure. I'll update the code to check that.